### PR TITLE
[wip] feat: improve code tags display

### PR DESCRIPTION
**Pull request by dzuqe**
steps toward #1368

summary:
I've created a new nim object that composites highlite's functionality and decorates key
tokens with colored html spans

usage: just add `lang=nim` to the beginning of the code segment (or any of the following languages ["C", "c++", "jAvA", "Nim", "c#", "yaml", "python"])
\```
lang=nim
import strformat

type
  Person = object
    name: string
    age: Natural # Ensures the age is positive
\```

future considerations:
- a better theme system that doesn't require usage of the tenary operator to get the colors right for both themes
- respect the tabs when pasting code in the chat input

caveats:
there was an instance where i chose the language to be c++ whereas the
code embedded wasn't c++ and the code crashed. However, i have not been
able to reproduce this again

2021-04-02-123104_841x426_scrot

2021-04-02-123116_668x360_scrot

### CODE REVIEW

**Comment by michaelsbradleyjr** [comment]
Very nice!

In GitHub and Discord, for example, the name of the lang is specified right after the opening triple backticks, like this:

image

Would it be possible to support that instead of having a `lang=...` line?

**Comment by rinzlxr** [comment]
Thanks! 
That would require altering how the model layer (status-go/nim-waku) stores and sends messages back to us. 

currently, the first line containing backticks gets deleted (including anything after) when the message is retrieved. That's why i use `lang=nim` on the second line instead.

**Comment by richard-ramos** [comment]
Once https://github.com/status-im/markdown/pull/3 gets merged, and added into status-go,  `p.Msg` in `src/app/chat/views/message_format.nim` L35  will contain an attribute `language` which can be used instead of having to specify the language in the second line

**Comment by richard-ramos** [comment]
This PR updates the version of markdown module used in status-go to include the language attribute. 
https://github.com/status-im/status-go/pull/2188

**Comment by rinzlxr** [comment]
Hey thanks Richard!

I updated the `TextItem` type to include the language attribute
2021-04-06-151021_331x128_scrot

Do you know if there's some unmarshalling method that would require changes too? (ie to include the new attribute in the method?)

**Comment by richard-ramos** [comment]
Yes, there's an object that should be modified to add the language attribute

https://github.com/status-im/status-desktop/blob/master/src/status/chat/message.nim#L17-L21
https://github.com/status-im/status-desktop/blob/master/src/status/signals/messages.nim#L221-L227

**Comment by richard-ramos** [comment]
The version of status-go we are currently pointing to should be updated as well:
```
cd vendor/status-go
git pull
git checkout update-markdown
```

File: ui/imports/Utils.qml [hunk 1]
```diff
@@ -50,12 +50,18 @@ QtObject {
     }
 
     function getMessageWithStyle(msg, useCompactMode, isCurrentUser) {
+        let fMsg = msg.startsWith('<code>') ? `<div class='code-wrapper'>${msg}</div>` : msg
+        const codeBg = Style.current.name === 'dark' ? Style.current.darkGrey : Style.current.lightGrey
```
Comment (jrainville) [inline ui/imports/Utils.qml:54]: We can just add a new Style property for `codeBg`, that way you don't need the ternary here.

**Review by jrainville — COMMENTED** [review]
Very nice job!

**Comment by rinzlxr** [comment]
I updated the relevant unmarshalling proc to register the `language` attribute in the `TextItem` type. 
However, I am blocked by understanding whether the message sent from the chat is stored correctly with the new attribute in the model layer.
I am yet to understand how I can setup a simulated local test environment so that I am able to catch breakpoints on the status-go side/waku side.
I am opening a new issue that describes what's blocking me, maybe "more eyes on the problem" can manifest a quicker solution

**Comment by iurimatias** [comment]
closing due to inactivity, draft state

**Pull request closed without merging**